src/main/index.js

```diff
@@ -175,6 +181,18 @@ export function queuePrint(printSpec) {
     .toPromise();
 }
 
+/**
+ * Starts generating a legend image from a print spec.
+ * @param {PrintSpec} printSpec
+ * @return {Promise<Blob>} Promise resolving to the final legend image blob.
+ */
+export function createLegends(printSpec) {
+  const spec = printSpec.value ? printSpec.value : printSpec;
+  if (spec.layers.find(el => el.legend)) {
+    return getLegends(spec);
+  }
+}
```

Comment:
Maybe it would make sense here to put a warning in the console saying that no layers had legends enabled? Otherwise it might end up not doing anything and that could be confusing for the user